workflow/docker_util.py

```diff
@@ -0,0 +1,201 @@
+    params_base32 = base64.b32encode(params_hash.digest()).decode('ascii')
+    if length is None or length < 1 or length > len(params_base32):
+        return params_base32
+    else:
+        return params_base32[:length]
+
+class DockerClient(object):
+    def __init__(self, client: str, 
+                workflow_settings: Optional[WorkflowRun] = None, 
+                cytoscape: Optional[Visualize] = None) -> None:
+
+        self.client_name = client
+        self.client = CLIENTS[client]
+        self.workflow_runner = workflow_settings
+        self.outfolder = workflow_settings.out_folder
+        self.cytoscape = cytoscape
+        self.DOCKER_SETTINGS = {
+            'TPS'       : self.prepare_tps,
+            'CYTOSCAPE'  : self.prepare_cytoscape
+        }
+
+    def _collect_files(self) -> List[os.PathLike]:
+        self.files_ = []
+        for _ , path in self.workflow_runner.tps_input_settings.args.items():
+
+            try:
+                temp_file = Path(path)
+                if temp_file.is_file():
+                    self.files_.append(temp_file)
+                else:
+                    print(f"{path} not path")
+            except TypeError:
+                print('not path')
+
+        temp_dir = tempfile.mkdtemp(dir=Path('data'))
+
+    def create_temporary_copy(self, path):
+        temp_dir = tempfile.gettempdir()
+        print(temp_dir)
+        temp_path = os.path.join(temp_dir, 'temp_file_name')
+        shutil.copy2(path, temp_path)
+        return temp_path
+
+    def _create_output_dir(self) -> None:
+        '''
+        generate output tag: output_tag_
+
+        '''
+        check_build = {
+            self.workflow_runner.build[i]: self.workflow_runner.build[i+1] \
+            for i in range(0, len(self.workflow_runner.build), 2)
+        }
+
+        if '--outfolder' in check_build:
+            print('override outfolder location')
+            del check_build['--outfolder']
+            self.workflow_runner.build = list(functools.reduce(lambda x, y: x+y, check_build.items()))
+
+        hash_val = hash_params_sha1_base32(self.workflow_runner.build, length=DEFAULT_HASH_LENGTH)
+        self.output_tag_ = DOCKER_OUTFOLDER.format(hash_val)
+        self.workflow_runner.build.extend(['--outfolder', self.output_tag_])
+        self.workflow_runner.out_folder = self.output_tag_
+
+    def prepare_tps(self) -> Dict[str, Any]:
+        """Helper fucntion for preparing TPS Container Settings
+
+        Returns:
+            Dict[str, Any]: collection of Docker Settings
+        """
+        self._create_output_dir()
+        container_settings = {
+            'work_dir':None,
+            'command':None,
+            'volumes': Dict[str, Dict[str, Any]],
+            'container_wdir':None,
+        }
+
+        work_dir = Path(__file__).parent.parent.absolute()
+        data_path = prepare_path_docker(work_dir)+'/data'
+        output_path = prepare_path_docker(work_dir)+'/'+self.output_tag_

```

Comment:
Will these paths work on different platforms?

Reply:
In theory they should. These paths will only be used in the docker containers